src/app/components/POCs/ProcessorEdit/ProcessorEdit.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useCallback, VoidFunctionComponent } from "react";
+import { useMachine } from "@xstate/react";
+import { PageSection, PageSectionVariants } from "@patternfly/react-core";
+import ProcessorMachine from "@app/components/POCs/ProcessorEdit/ProcessorMachine";
+import ProcessorEditHeader from "@app/components/POCs/ProcessorEditHeader/ProcessorEditHeader";
+import ProcessorCodeEditor from "@app/components/POCs/ProcessorCodeEditor/ProcessorCodeEditor";
+import { ProcessorTemplateSelector } from "@app/components/POCs/ProcessorTemplateSelector/ProcessorTemplateSelector";
+import { ProcessorTemplate } from "@app/components/POCs/ProcessorEdit/ProcessorTemplates";
+import DeployBlockedModal from "@app/components/POCs/ProcessorEdit/components/DeployBlockedModal";
+import { useTranslation } from "@rhoas/app-services-ui-components";
+import "./ProcessorEdit.css";
+
+export interface ProcessorEditProps {
+  /** List of processor templates */
+  processorTemplates: ProcessorTemplate[];
+  /** List of sink values to use as suggestions while editing */
+  sinkValuesSuggestions: string[];
+  /** Callback to create a processor */
+  createProcessor: (
+    data: {
+      processorName: string;
+      code: string;
+    },
+    onSuccess: () => void,
+    onError: () => void
+  ) => void;
+  /** Callback to cancel the creation flow */
+  onCancel: () => void;
+  /** Existing name of the processor */
+  existingProcessorName?: string;
+  /** Existing code of the processor */
+  existingCode?: string;
+}
+
+const ProcessorEdit: VoidFunctionComponent<ProcessorEditProps> = (props) => {
+  const {
+    processorTemplates,
+    sinkValuesSuggestions,
+    createProcessor,
+    onCancel,
+    existingCode,
+    existingProcessorName,
+  } = props;
+  const { t } = useTranslation("smartEventsTempDictionary");
+
+  const [current, send] = useMachine(ProcessorMachine, {
+    context: {
+      processorName: existingProcessorName ?? t("processor.processorName"),
+      code: existingCode ?? processorTemplates[0].code,
+      templates: processorTemplates,
+      selectedTemplate: processorTemplates[0].id,
+      sinkSuggestions: sinkValuesSuggestions,
+    },
+    services: {
+      createProcessor: (context) => {
+        const { code, processorName } = context;
+        return (send) => {
+          function onSuccess(): void {
+            send("create success");
+          }
+          function onError(): void {
+            send("create error");
+          }
```

Comment:
APIs errors are not yet fully managed. I'll take care of them when adding the processor creation to the UI and plugging the APIs.